### PR TITLE
[deconz] Added util channels (e.g. 'battery')

### LINKED ISSUE #4524: [deconz] Add "dark" channel from Ikea Trådfri motion sensor
The IKEA Trådfri motion sensor provides a "dark" boolean attribute at config.dark. It would be nice to have that as a channel. Also battery level would be a nice to have. See below JSON from my motion sensor

`{
"config":{
"alert": "none",
"battery": 60,
"delay": 60,
"duration": 60,
"group": "53372",
"on": true,
"reachable": true
},
"ep": 1,
"etag": "986612993cac4328153ab129c9ce2212",
"manufacturername": "IKEA of Sweden",
"modelid": "TRADFRI motion sensor",
"name": "TRADFRI motion sensor ",
"state":{
"dark": true,
"lastupdated": "2019-01-05T20:38:59",
"presence": false
},
"swversion": "1.2.214",
"type": "ZHAPresence",
"uniqueid": "d0:cf:5e:ff:fe:12:4c:0e-01-0006"
}`

### LINKED ISSUE #4538: [deconz] Add "battery" channel
The battery status of things is already displayed as a property. Unfortunately you can't use this in rules, e.g. to send a low battery notification to your smartphone.

It would be nice if the battery status were available as a channel.

### PR DESCRIPTION
**Pull request by cweitkamp**
- Added util channels (e.g. `battery_level`, `battery_low` and `last_updated`)
- Added more light sensor channels (e.g. `lightlevel`, `dark` and `daylight`)
- Added check for the optional `dark` channel on presence sensors

Test version:
[org.openhab.binding.deconz-2.5.0-SNAPSHOT.jar.zip](https://github.com/openhab/openhab2-addons/files/2746191/org.openhab.binding.deconz-2.5.0-SNAPSHOT.jar.zip)

~I did not find the time to do a real test. Just threw in the code.~

Closes #4538
Closes #4524

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: addons/binding/org.openhab.binding.deconz/README.md [hunk 1]
```diff
@@ -65,27 +67,30 @@ Bridge deconz:deconz:homeserver [ host="192.168.0.10", apikey="ABCDEFGHIJ" ]
...
+| temperature     | Number:Temperature   |      R      | Current temperature in ˚C                                                 | temperaturesensor          |
+| humidity        | Number:Dimensionless |      R      | Current humidity in %                                                     | humiditysensor             |
+| open            | Contact              |      R      | Status of contacts: `OPEN`; `CLOSED`                                      | openclosesensor            |
+| light           | String               |      R      | Light level: `Daylight`,`Sunset`,`Dark`                                   | daylightsensor             |
+| value           | Number               |      R      | Sun position: `130` = dawn; `140` = sunrise; `190` = sunset; `210` = dusk | daylightsensor             |
+| battery_level   | Number               |      R      | Battery level (in %)                                                      | any battery-powered sensor |
+| battery_low     | Number               |      R      | Battery level low: `ON`; `OFF`                                            | any battery-powered sensor |
```
Comment (Markkuuss) [inline addons/binding/org.openhab.binding.deconz/README.md]:
```suggestion
| battery_low     | Switch                |      R      | Battery level low: `ON`; `OFF`                                            | any battery-powered sensor |
```

**Comment by cweitkamp** [comment]
As we do not receive an event from the websocket for changes in the `config` section, I expect the values of the battery channels will not change immediately - only if a call for the "initial data" will be performed. Shall we add a note about that in the documentation?

**Comment by cweitkamp** [comment]
I did my tests now and added a minor fix to finalize this PR.

File: addons/binding/org.openhab.binding.deconz/ESH-INF/thing/thing-types.xml [hunk 2]
```diff
@@ -136,6 +139,27 @@
...
 	</channel-type>
 
+	<channel-type id="light_level" advanced="true">
+		<item-type>Number</item-type>
+		<label>Light Level</label>
+		<description>Current light level.</description>
+		<state readOnly="true" pattern="%d" />
+	</channel-type>
+
+	<channel-type id="dark">
```
Comment (davidgraeff) [inline addons/binding/org.openhab.binding.deconz/ESH-INF/thing/thing-types.xml]: I think I do not understand the last commit. We now have "dark", "daylight", "light_level", "lightlux"?

Comment (cweitkamp) [reply addons/binding/org.openhab.binding.deconz/ESH-INF/thing/thing-types.xml]: Those four channels are present for a light sensor (see https://github.com/dresden-elektronik/deconz-rest-plugin/wiki/Supported-Devices#measured-values-legend). Three of them are part of the Philips Hue API too. Lux is missing. From Philips Hue perspective: `dark` and `daylight` are calculated flags based on two user-defined thresholds. In the Hue APP you are able to set those values, the lower one defines threshold for darkness, the upper one defines threshold for daylight. In-between is "basic" / "indoor" illumination. Philips is working with lightlevel, whereas others prefer lux. From my point of view the lightlevel is useless thus it is advanced.

Comment (davidgraeff) [reply addons/binding/org.openhab.binding.deconz/ESH-INF/thing/thing-types.xml]: k, thanks for the explanation

**Comment by martinvw** [comment]
@cweitkamp can you rebase :-)

**Comment by cweitkamp** [comment]
Will do that. Give me a minute.

**Comment by cweitkamp** [comment]
Squashed and rebased.

**Comment by martinvw** [comment]
Thanks!

**Pull request merged**